ebpf-profiling.graphqls

```diff
@@ -0,0 +1,172 @@
+    onCPUTarget: EBPFProfilingOnCPUTarget
+}
+
+# eBPF Profiling task create result
+type EBPFProfilingTaskCreationResult {
+    # if null or empty means the task create success, otherwise get create error reason
+    errorReason: String
+
+    # get task id when create success
+    id: String
+}
+
+# query eBPF profiling task condition
+input EBPFProfilingTaskCondition {
+    # the process finder type of profiling task
+    finderType: EBPFProfilingProcessFinderType
+    # service id of process which need profiling
+    serviceId: ID
+    # instance id of process which need profiling
+    instanceId: ID
+    # instance id of process which need profiling
+    processId: ID
+}
+
+# eBPF profiling task data
+type EBPFProfilingTask {
+    # profiling task id
+    taskId: ID!
+    # profiling process finder type
+    processFinderType: EBPFProfilingProcessFinderType!
+    # service of profiling task
+    serviceId: ID
+    serviceName: String
+    # instance of profiling task
+    instanceId: ID
+    instanceName: String
+    # process of profiling task
+    processId: ID
+    processName: String
+
+    # profiling task start timestamp
+    taskStartTime: Long!
+    # profiling task trigger type
+    triggerType: EBPFProfilingTriggerType!
+    # "FIXED_TIME" type task profiling duration
+    fixedTriggerDuration: Long
+
+    # profiling task target type
+    targetType: EBPFProfilingTargetType!
+    # "ON_CPU" type task profiling dump period
+    onCPUDumpPeriod: Int
+
+    # profiling task create timestamp
+    createTime: Long!
+}
+
+type EBPFProfilingSchedule {
+    # profiling task schedule id
+    scheduleId: ID!
+    # profiling task id
+    taskId: ID!
+    # process id
+    processId: ID!
+    # profiling schedule start timestamp
+    startTime: Long!
+    # profiling schedule finished timestamp
+    endTime: Long!
+}
+
+input EBPFProfilingAnalyzeTimeRange {
+    # start timestamp
+    start: Long!
+    # end timestamp
+    end: Long!
+}
+
+type EBPFProfilingAnalyzation {
+    # if not empty means backend has information gave to the user
+    tip: String
+    # profiling analyzed trees
+    trees: [EBPFProfilingTree]!
+}
+
+type EBPFProfilingTree {
+    # profiling stack elements
+    elements: [EBPFProfilingStackElement!]!
+}
+
+type EBPFProfilingStackElement {
+    # the element id
+    id: Int!
+    # the parent element id, working on build the tree
+    parentId: Int!
+    # stack element symbol name
+    symbol: String!
+    # stack element type
+    stackType: EBPFProfilingStackType!
+    # current stack element total dump count
+    dumpCount: Long!
+}
+
+enum EBPFProfilingStackType {
+    KERNEL_SPACE,
+    USER_SPACE
+}
+
+enum EBPFProfilingProcessFinderType {
+    PROCESS_ID
+}
+
+enum EBPFProfilingTriggerType {
+    FIXED_TIME
+}
+
+enum EBPFProfilingTargetType {
+    ON_CPU
+}
+
+input EBPFProfilingFixedTimeTrigger {
+    duration: Int!
+}
+
+input EBPFProfilingOnCPUTarget {
+    dumpPeriod: Int!
+}
```

Comment:
These seem to belong to the rover. The ebpf profiling is much more OS level than SDK profiling. I prefer not exposing all things from UI side.

Reply:
Deleted. The dump count will be added as a configuration item on the rover side.